src/coreclr/jit/lsra.cpp

```diff
@@ -11264,7 +11264,8 @@ void LinearScan::RegisterSelection::try_SPILL_COST()
         RefPosition* recentRefPosition  = spillCandidateRegRecord->assignedInterval != nullptr
                                              ? spillCandidateRegRecord->assignedInterval->recentRefPosition
                                              : nullptr;
-        if ((recentRefPosition != nullptr) && (recentRefPosition->RegOptional() &&
+        if ((recentRefPosition != nullptr) &&
```

Comment:
My bad for not spotting this formatting error in https://github.com/dotnet/runtime/pull/55247. I also missed one more change to hoist `spillCandidateRegRecord->assignedInterval`. Do you mind changing that as well?

```c++
Interval* assignedInterval = spillCandidateRegRecord->assignedInterval;

// and then use assingedInterval at multiple places.
```
```

Reply:
done

Reply:
You might need to define it after the line:

```c++
RegRecord* spillCandidateRegRecord = &linearScan->physRegs[spillCandidateRegNum];
Interval*    assignedInterval      = spillCandidateRegRecord->assignedInterval;
```

so that you can use it in:

```c++
if ((linearScan->getNextIntervalRef(spillCandidateRegNum, regType) == currentLocation) &&
            !assignedInterval->getNextRefPosition()->RegOptional())
        {
            continue;
        }
```

Reply:
could you please open a PR with a fix? I think it would be faster and less confusing for the history/blame.